docs/source/isolation.rst

```diff
@@ -0,0 +1,188 @@
+==============================================
+Resource Isolation in PaaSTA, Mesos and Docker
+==============================================
+
+PaaSTA instance definitions include fields that specify the required resources
+for your service. The reason for this is two-fold: firstly, so that whichever
+Mesos framework (Marathon, Chronos) can evaluate which Mesos agent making
+offers have enough capacity to run the task (and pick one of the agents
+accordingly); secondly, so that tasks can be protected from especially noisy
+neighbours on a box. That is, if a task under-specifies the resources it
+requires to run, or in another case, has a bug that means that it consumes far
+more resources than it *should* require, then the offending tasks can be
+isolated effectively, preventing them from having a negative impact on it's
+neighbours.
+
+This document is designed to give a more detailed review of how Mesos
+Frameworks such as Marathon and Chronos use these requirements to run tasks on
+different Mesos agents, and how these isolation mechanisms are implemented.
+
+Note: Knowing the details of these systems isn't a requirement of using PaaSTA;
+most service authors may never need to know the details of such things. In
+fact, one of PaaSTA's primary design goals is to *hide* the minutiae of
+schedulers and resource isolation. However, this may benefit administrators
+of PaaSTA (and, more generally, Mesos clusters), and the simply curious.
+
+Final note: The details herein may, nay, will contain (unintended) inaccuracies.
+If you notice such a thing, we'd be super grateful if you could open a pull
+request to correct the document!
+
+How Tasks are Scheduled on Hosts
+--------------------------------
+
+To first understand how these resources are used, one must understand how
+a task is run on a Mesos cluster.
+
+Mesos can run in two modes: Master and Agent. When a node is running Mesos in
+Master mode, it is responsible for communicating between agent processes and
+frameworks. A Framework is a program which wants to run tasks on the Mesos
+cluster.
+
+A master is responsible for presenting frameworks with resource offers.
+Resource offers are compute resource free for a framework to run a task. The
+details of that compute resource comes from the agent nodes, which regularly
+tell the Master agent the resources it has available for running tasks. Using
+the correct parlance, Mesos agents make 'offers' to the master.
+
+At Yelp, the resource that a given agent has available is determined by a
```

Comment:
Is it possible to helpfully make this a little more generic? Like, "If the box is doing things other than paasta then you can run a script to... We do this on some of our service nodes at Yelp."